### DISCUSSION TITLE
Reinforcement Learning

### DISCUSSION THREAD

**Discussion by sesem738** [bitcraze/discussions#938] [Q&A]
Hello,

I have acquired a Crayflie AI Deck. I am very new to drones in general. The goal is to implement reinforcement learning based navigation. I also want to use Optitrack for ground truth pose data. I wanted to ask if there are any resources that can help me going.

**Comment by krichardsson** [comment]
Hi,

Some random pointers

An overview of the system is available at https://www.bitcraze.io/documentation/system/

There are tutorials of how to get started with the Crazyflie and AI-deck in https://www.bitcraze.io/documentation/tutorials/

There is an example of how to connect to mocap systems (including optitrack) in https://github.com/bitcraze/crazyflie-lib-python/blob/master/examples/mocap/mocap_hl_commander.py

If you are interested in using simulation for your reinforcement learning, check out #779

You can find source code and examples for various parts of the system in https://www.bitcraze.io/documentation/repository/ ,
including GAP8 examples.

**Reply by sesem738** [reply]
Has anyone done sim-to-real with the webots simulation?

**Reply by knmcguire** [reply]
Yes, they have. I can't find a paper for you now but I know that several teams of the IMAV 2022 nano quadcopter challenge used webots to pretrain their networks with webots https://www.youtube.com/live/WaDU4I2TImA?si=mPDZjbLvcm6RkMpi. [Cyberbotics ](https://www.cyberbotics.com/)is also known for focusing for sim-to-real applications as well (they made webots). Perhaps ask on their community forum as well if they know of any reinforcement application.
